modules/geometry-calculation.js

```diff
@@ -18,16 +38,20 @@ const calculateJunctionPoint = (line1, line2) => {
     }
     else {
         const x = (line1.n - line2.n) / (line2.slope - line1.slope)
-        const junctionPoint = line1.getPointByX(x);
+        const junctionPoint=line1.getPointByX(x);
         return junctionPoint
     }
 }
 
 const isPointOnLine = (line, point) => {
+    if (!(line instanceof Line) || !(point instanceof Point)) {
+        throw Error('The value is of an invalid type')
+    }
+
     const proxyLine = new Line({ point1: line.point1, point2: point })
-    proxyLine.calculateSlope()
+    proxyLine.slope=proxyLine.calculateSlope()
     if (line.slope === proxyLine.slope) {
-        proxyLine.calculateNOfLineFunction()
+        proxyLine.n=proxyLine.calculateNOfLineFunction()
```

Comment:
the same here, the n will be updated into undefined

if you did it for the mocks, in the mocks you can do whatever you want, but here you need to use the function as it has to be